aip/0180.md

```diff
@@ -0,0 +1,126 @@
+---
+aip:
+  id: 180
+  state: reviewing
+  created: 2019-07-23
+permalink: /180
+redirect_from:
+  - /0180
+---
+
+# Backwards compatibility
+
+APIs are fundamentally contracts with users, and users often write code against
+APIs that is then launched into a production service with the expectation that
+it continues to work (unless the API has a [stability level][aip-181] that
+indicates otherwise). Therefore, it is important to understand what constitutes
+a backwards compatible change and what constitutes a backwards incompatible
+change.
+
+## Guidance
+
+Existing client code **must not** be broken by a service updating to a new
+minor or patch release. Old clients **must** be able to work against newer
+servers (with the same major version number).
+
+**Important:** It is not always salient whether a change is compatible or not.
+The guidance here **should** be treated as indicative, rather than as a
+comprehensive list of every possible change.
+
+There are three distinct types of compatibility to consider:
+
+1. Source compatibility: Code written against a previous version **must**
+   compile against a newer version, and successfully run with a newer version
+   of the client library.
+2. Wire compatibility: Code written against a previous version **must** be able
+   to communicate correctly with a newer server. In other words, not only are
+   inputs and outputs compatible, but the serialization and deserialization
+   expectations continue to match.
+3. Semantic compatibility: Code written against a previous version **must**
+   continue to receive the intended and expected results.
+
+### Adding components
+
+In general, new components (interfaces, methods, messages, fields, enums, or
+enum values) **may** be added to existing APIs in the same major version.
+
+However, keep the following guidelines in mind when doing this:
+
+- Code written against the previous surface (and thus is unaware of the new
+  components) **must** continue to be treated the same way as before.
+  - New required fields **must not** be added to existing request messages or
+    resources.
+  - Any field being populated by clients **must** have a default behavior
+    matching the behavior before the field was introduced.
+  - Any field previously populated by the server **must** continue to be
+    populated, even if it introduces redundancy.
+- Enums **may** be freely added to request messages. For response messages and
```

Comment:
This bullet conflates adding new enum fields and adding new values to existing enums - at least in the way I'm reading it. If this is meant to be just about adding new values to enums, it needs to be clarified. Otherwise, let's split it into two bullets.

Reply:
This was meant to be just enum values. Fixed.